### PR TITLE
Add Dictionary test codes and modify Dictionaryobject route

### PR DESCRIPTION
**Pull request by miloul**
This PR is add Dictionary test code and modify dictionary object route to dictionaryobject

As the existing integration test case haven't covered the dictionary type, other than dictionary object, the PR wil cover the dictionary type. And modify dictionary object route from dictionary to dictionaryobject. Because dictionary trigger has route that name is dictionary.

### CODE REVIEW

File: test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp/Get_ApplicationJson_Dictionary_HttpTrigger.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            var result = new OkResult();
+
+            return await Task.FromResult(result).ConfigureAwait(false);
+        }
+
+        [FunctionName(nameof(Get_ApplicationJson_Dictionary_HttpTrigger.Get_ApplicationJson_Dictionary_I))]
+        [OpenApiOperation(operationId: nameof(Get_ApplicationJson_Dictionary_HttpTrigger.Get_ApplicationJson_Dictionary_I), tags: new[] { "dictionary" })]
+        [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, contentType: "application/json", bodyType: typeof(IDictionary<string,int>), Description = "The OK response")]
+        public static async Task<IActionResult> Get_ApplicationJson_Dictionary_I(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "GET", Route = "get-applicationjson-dictionary-i")] HttpRequest req,
```
Comment (justinyoo) [inline test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp/Get_ApplicationJson_Dictionary_HttpTrigger.cs:31]: Can you change the route value to be more relevant to the return dictionary object, as well as the method name?

eg) `get-applicationjson-dictionary-idictionary`, `Get_ApplicationJson_Dictionary_IDictionary`

File: test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp/Get_ApplicationJson_Dictionary_HttpTrigger.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            var result = new OkResult();
+
+            return await Task.FromResult(result).ConfigureAwait(false);
+        }
+
+        [FunctionName(nameof(Get_ApplicationJson_Dictionary_HttpTrigger.Get_ApplicationJson_Dictionary_Ireadonly))]
+        [OpenApiOperation(operationId: nameof(Get_ApplicationJson_Dictionary_HttpTrigger.Get_ApplicationJson_Dictionary_Ireadonly), tags: new[] { "dictionary" })]
+        [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, contentType: "application/json", bodyType: typeof(IReadOnlyDictionary<string,double>), Description = "The OK response")]
+        public static async Task<IActionResult> Get_ApplicationJson_Dictionary_Ireadonly(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "GET", Route = "get-applicationjson-dictionary-ireadonly")] HttpRequest req,
```
Comment (justinyoo) [inline test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp/Get_ApplicationJson_Dictionary_HttpTrigger.cs:43]: Can you change the route value to be more relevant to the return dictionary object, as well as the method name?

eg) `get-applicationjson-dictionary-ireadonlydictionary`, `Get_ApplicationJson_Dictionary_IReadOnlyDictionary`

File: test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp/Get_ApplicationJson_Dictionary_HttpTrigger.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            var result = new OkResult();
+
+            return await Task.FromResult(result).ConfigureAwait(false);
+        }
+
+        [FunctionName(nameof(Get_ApplicationJson_Dictionary_HttpTrigger.Get_ApplicationJson_Dictionary_Key))]
+        [OpenApiOperation(operationId: nameof(Get_ApplicationJson_Dictionary_HttpTrigger.Get_ApplicationJson_Dictionary_Key), tags: new[] { "dictionary" })]
+        [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, contentType: "application/json", bodyType: typeof(KeyValuePair<string,bool>), Description = "The OK response")]
+        public static async Task<IActionResult> Get_ApplicationJson_Dictionary_Key(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "GET", Route = "get-applicationjson-dictionary-key")] HttpRequest req,
```
Comment (justinyoo) [inline test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp/Get_ApplicationJson_Dictionary_HttpTrigger.cs:55]: Can you change the route value to be more relevant to the return dictionary object, as well as the method name?

eg) `get-applicationjson-dictionary-keyvalutpair`, `Get_ApplicationJson_Dictionary_KeyValuePair`

File: test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.Document.Tests/Get_ApplicationJson_Dictionary_Tests.cs [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        [DataRow("/get-applicationjson-dictionary", "get", "200", "application/json", "object", "string")]
+        [DataRow("/get-applicationjson-dictionary-i", "get", "200", "application/json", "object", "integer")]
+        [DataRow("/get-applicationjson-dictionary-ireadonly", "get", "200", "application/json", "object", "number")]
+        [DataRow("/get-applicationjson-dictionary-key", "get", "200", "application/json", "object", "boolean")]
+        public void Given_OpenApiDocument_Then_It_Should_Return_OperationResponseContentTypeSchemaPropertiesType(string path, string operationType, string responseCode, string contentType, string dataType, string itemType)
+        {
+            var content = this._doc["paths"][path][operationType]["responses"][responseCode]["content"];
+
+            var schema = content[contentType]["schema"];
+            var additionaltype =schema["additionalProperties"]["type"];
```
Comment (justinyoo) [inline test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.Document.Tests/Get_ApplicationJson_Dictionary_Tests.cs:102]: For readability, it's better to add a space before and after the equal sign.

File: test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.Document.Tests/Get_ApplicationJson_Dictionary_Tests.cs [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+            additionaltype.Value<string>().Should().Be(itemType);
```
Comment (justinyoo) [inline test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.Document.Tests/Get_ApplicationJson_Dictionary_Tests.cs:103]: Can you please add one blank line before this line?

**Review by justinyoo — CHANGES_REQUESTED** [review]
@miloul Thanks for your PR! I left a few comments for you to take further look.

File: test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp/Get_ApplicationJson_Dictionary_HttpTrigger.cs (hunk 1 above)
Comment (miloul) [reply test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp/Get_ApplicationJson_Dictionary_HttpTrigger.cs]: Updated!

File: test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp/Get_ApplicationJson_Dictionary_HttpTrigger.cs (hunk 2 above)
Comment (miloul) [reply test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp/Get_ApplicationJson_Dictionary_HttpTrigger.cs]: Updated!

File: test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp/Get_ApplicationJson_Dictionary_HttpTrigger.cs (hunk 3 above)
Comment (miloul) [reply test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp/Get_ApplicationJson_Dictionary_HttpTrigger.cs]: Updated!

File: test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.Document.Tests/Get_ApplicationJson_Dictionary_Tests.cs (hunk 4 above)
Comment (miloul) [reply test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.Document.Tests/Get_ApplicationJson_Dictionary_Tests.cs]: Updated!

File: test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.Document.Tests/Get_ApplicationJson_Dictionary_Tests.cs (hunk 5 above)
Comment (miloul) [reply test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.Document.Tests/Get_ApplicationJson_Dictionary_Tests.cs:105]: Updated!

**Review by justinyoo — APPROVED** [review]
Thanks! LGTM!

**Pull request merged**